pkg/datapath/loader/base.go

```diff
@@ -424,18 +424,18 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 	ctx, cancel := context.WithTimeout(ctx, defaults.ExecTimeout)
 	defer cancel()
 
-	extraArgs := []string{"-Dcapture_enabled=0"}
-	if err := l.reinitializeXDPLocked(ctx, extraArgs); err != nil {
-		log.WithError(err).Fatal("Failed to compile XDP program")
-	}
-
 	prog := filepath.Join(option.Config.BpfDir, "init.sh")
 	cmd := exec.CommandContext(ctx, prog, args...)
 	cmd.Env = bpf.Environment()
 	if _, err := cmd.CombinedOutput(log, true); err != nil {
 		return err
 	}
 
+	extraArgs := []string{"-Dcapture_enabled=0"}
+	if err := l.reinitializeXDPLocked(ctx, extraArgs); err != nil {
+		log.WithError(err).Fatal("Failed to compile XDP program")
+	}
```

Comment:
Any particular reason why you moved this down, or just some clean-up?

Reply:
Ah, that was a separate commit initially - must have accidentally squashed at some point :(.

`init.sh` adds TUNNEL_MODE to the `node_config.h` header file. So I think we need to delay building the XDP program if we want to rely on that macro.

Reply:
Sounds good, please consider mentioning that in a comment or commit log

Reply:
updated the commit msg